nsis/src/nsis/NsisLoader.java

```diff
@@ -128,17 +124,18 @@ protected void load(ByteProvider provider, LoadSpec loadSpec, List<Option> optio
 	 * @throws CancelledException
 	 * @throws DuplicateNameException
 	 * @throws LockException
-	 * @throws CodeUnitInsertionException 
+	 * @throws CodeUnitInsertionException
 	 */
-	private void initScriptHeader(FileBytes fileBytes, Address scriptHeaderAddress, long size,
-			Program program, DataType dataType) throws MemoryConflictException,
-			AddressOverflowException, CancelledException, DuplicateNameException, LockException, CodeUnitInsertionException {
+	private void initScriptHeader(InputStream is, Address scriptHeaderAddress, Program program,
+			DataType dataType, TaskMonitor monitor, int size)
+			throws MemoryConflictException, AddressOverflowException, CancelledException,
+			DuplicateNameException, LockException, CodeUnitInsertionException {
 		Memory memory = program.getMemory();
-		MemoryBlock new_block = memory.createInitializedBlock(".script_header", scriptHeaderAddress,
-				fileBytes, 0, size, false);
-		new_block.setRead(true);
-		new_block.setWrite(false);
-		new_block.setExecute(false);
+		MemoryBlock scriptHeaderBlock = memory.createInitializedBlock(".script_header",
+				scriptHeaderAddress, is, size, monitor, false);
```

Comment:
just double-checking here: Does MemoryBlock need `InputStream is` to remain open or can it be closed after call `createInitializedBlock` is done? 
(The stream is closed in lines 94-97 above)
(I'm just a C++ developer confused by resource ownership in Java)

Reply:
From what I understand, the InputStream can be closed because the MemoryBlock uses it to initialize a memory block, once it is initialized the data is saved elsewhere and the InputStream can be closed

Reply:
Ack